embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/capture/crash/EmbraceCrashService.kt

```diff
@@ -76,14 +78,15 @@ internal class EmbraceCrashService(
             // is enabled for an Unity build. When a native crash occurs and the NDK sends an
             // uncaught exception the SDK assign the unity crash id as the java crash id.
             val unityCrashId = ndkService.getUnityCrashId()
+            val crashNumber = preferencesService.incrementAndGetCrashNumber()
```

Comment:
Should we do a best-effort only here in case the preferences service is unavailable or throws an exception? The fallback for the `Future.get()` seems to be doing it on the calling thread which could fail too.

Reply:
Maybe we should add a try catch here, as it's used by every "incrementAndGet" function: 

```
private fun incrementAndGetOrdinal(key: String): Int {
        val ordinal = (prefs.getIntegerPreference(key) ?: 0) + 1
        prefs.setIntegerPreference(key, ordinal)
        return ordinal
}
```

Right now, the session number doesn't have any error handling, so we could solve some issues if we add it here

Reply:
Try catch and return `-1` or some other value that well tell us an error has occurred while incrementing seems like a good idea